ShopifySharp/Lists/LinkHeaderParser.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace ShopifySharp.Lists
+{
+    public static class LinkHeaderParser
+    {
+        private static Regex _regexPrevLink = new Regex(@"<(https://[^>]*)>\s*;\s*rel=""previous""", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        private static Regex _regexNextLink = new Regex(@"<(https://[^>]*)>\s*;\s*rel=""next""", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        public static LinkHeaderParseResult Parse(string linkHeaderValue)
+        {
+            var prevLink = GetPageInfoParam(linkHeaderValue, _regexPrevLink);
+            var nextLink = GetPageInfoParam(linkHeaderValue, _regexNextLink);
+
+            if (prevLink == null && nextLink == null)
+                throw new ShopifyException($"Found neither a 'previous' or 'next' url in the link header: '{linkHeaderValue}'");
```

Comment:
I'm not sure if throwing an exception is a good idea yet. I haven't tested this, but I could imagine a scenario where you request the first page of e.g. orders, and your shop only has 1 total order so there's no next page or previous page to link to.

Reply:
I hesitated on this one however I left it in the end because Shopify stated that if there are no more pages, then the Link header will not be present at all.
So, if a Link header is present, it MUST have either a previous or next page.
Let me know if you still want to remove, but IMO the check is useful as it asserts that the parsing was succesful.

Reply:
Ah okay, I didn't know that! I think it's fine to leave it in then.

Reply:
cool